operator/controllers/btpoperator_controller_test.go

```diff
@@ -152,34 +260,61 @@ var _ = Describe("BTP Operator controller", func() {
 		})
 
 		It("soft delete (after timeout) should succeed", func() {
-			reconciler.SetTimeout(testTimeout)
-			reconciler.Client = newFakeK8s(reconciler.Client)
-
-			triggerDelete()
+			Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: testNamespace, Name: btpOperatorName}, cr)).To(Succeed())
+			Expect(k8sClient.Delete(ctx, cr)).To(Succeed())
+			Eventually(getCurrentCrStatus).
+				WithTimeout(stateChangeTimeout).
+				WithPolling(crStatePollingInterval).
+				Should(SatisfyAll(HaveField("State", types.StateDeleting)), HaveField("Conditions", HaveLen(1)))
+			Eventually(isCrNotFound).WithTimeout(deleteTimeout).WithPolling(crStatePollingInterval).Should(BeTrue())
 			doChecks()
 		})
 
 		It("soft delete (after hard deletion fail) should succeed", func() {
-			reconciler.SetTimeout(time.Minute * 1)
 			reconciler.Client = newFakeK8s(reconciler.Client)
 
-			triggerDelete()
+			Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: testNamespace, Name: btpOperatorName}, cr)).To(Succeed())
+			Expect(k8sClient.Delete(ctx, cr)).To(Succeed())
+			Eventually(getCurrentCrStatus).
+				WithTimeout(stateChangeTimeout).
+				WithPolling(crStatePollingInterval).
+				Should(SatisfyAll(HaveField("State", types.StateDeleting)), HaveField("Conditions", HaveLen(1)))
+			Eventually(isCrNotFound).WithTimeout(deleteTimeout).WithPolling(crStatePollingInterval).Should(BeTrue())
 			doChecks()
 		})
 
 		It("hard delete should succeed", func() {
-			reconciler.SetTimeout(time.Minute * 1)
-
+			Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: testNamespace, Name: btpOperatorName}, cr)).To(Succeed())
+			Expect(k8sClient.Delete(ctx, cr)).To(Succeed())
+			Eventually(getCurrentCrStatus).
+				WithTimeout(stateChangeTimeout).
+				WithPolling(crStatePollingInterval).
+				Should(SatisfyAll(HaveField("State", types.StateDeleting)), HaveField("Conditions", HaveLen(1)))
```

Comment:
I got `[PANICKED!] BTP Operator controller Deprovisioning [It] soft delete (after timeout) should succeed` while running the tests. `SatisfyAll` should contain both `HaveField` matchers, parenthesis encloses just one

Reply:
Corrected in the next commits.